jsonpickle/unpickler.py

```diff
@@ -607,23 +607,26 @@ def restore_key(key):
 
         return restore_key
 
-    def _restore_from_dict(self, obj, instance, ignorereserved=True):
+    def _restore_from_dict(self, obj, instance, restore_dict_items=True):
         restore_key = self._restore_key_fn()
         method = _obj_setattr
         deferred = {}
 
         for k, v in util.items(obj):
             # ignore the reserved attribute
-            if ignorereserved and k in tags.RESERVED:
+            if restore_dict_items and k in tags.RESERVED:
                 continue
             if isinstance(k, numeric_types):
                 str_k = k.__str__()
             else:
                 str_k = k
             self._namestack.append(str_k)
-            k = restore_key(k)
-            # step into the namespace
-            value = self._restore(v)
+            if restore_dict_items:
+                k = restore_key(k)
+                # step into the namespace
+                value = self._restore(v)
+            else:
+                value = v
```

Comment:
It looks like we're no longer calling `k = restore_key(k)` in the `else` block here. I haven't dug in deep enough to know whether or not this is an oversight.

Do all the tests still pass if we keep that?

Reply:
That is intentional to prevent double-decoding of dict keys - e.g. otherwise this would error:

```python
import jsonpickle

class Foo:
    def __init__(self):
        setattr(self, 'json://123', 'test')
    def __getstate__(self):
        return self.__dict__

f = Foo()
encoded = jsonpickle.encode(f, keys=True)
f2 = jsonpickle.decode(encoded, keys=True)
```

Reply:
This looks good to me!